src/datanode/src/server.rs

```diff
@@ -115,4 +117,16 @@ impl Services {
             .context(StartServerSnafu)?;
         Ok(())
     }
+
+    pub async fn shutdown(&self) -> Result<()> {
+        let mut res = vec![self.grpc_server.shutdown()];
+        if let Some(mysql_server) = &self.mysql_server {
+            res.push(mysql_server.shutdown());
+        }
+
+        futures::future::try_join_all(res)
+            .await
+            .context(ShutdownServerSnafu)?;
+        Ok(())
```

Comment:
```suggestion
        futures::future::try_join_all(res)
            .await
            .context(ShutdownServerSnafu)
```

Reply:
> Suggested change
>         futures::future::try_join_all(res)
>             .await
>             .context(ShutdownServerSnafu)?;
>         Ok(())
>         futures::future::try_join_all(res)
>             .await
>             .context(ShutdownServerSnafu)

It seems like leading to a mismatched types error:
```
mismatched types
expected enum `std::result::Result<(), _>`
   found enum `std::result::Result<Vec<()>, _>`
```